src/helpers/optionalStacking/common.ts

```diff
@@ -49,3 +52,66 @@ export function getSanitizedStackingConfig(
     }
     return stackingConfig;
 }
+
+export interface ISanitizedStackingConfig {
+    buckets: VisualizationObject.IBucket[];
+    config: IChartConfig;
+}
+
+/**
+ * Show a measure as a percentage
+ * one measure
+ *      => computeRatio = true (ignore stackBy)
+ *      => computeRatio = false or not set => evaluate stackMeasuresToPercent
+ * multiple measures
+ *      => ignore computeRatio and use stackMeasuresToPercent
+ *
+ * stackMeasures is applied only when there are [2 measures and more]
+ * stackMeasuresToPercent is applied only when there are [1 measure + 1 stackBy] or [2 measures and more]
+ */
+export function getSanitizedStackingConfigForChart(
+    buckets: VisualizationObject.IBucket[],
+    config: IChartConfig,
+): ISanitizedStackingConfig {
+    const newBuckets: VisualizationObject.IBucket[] = [...buckets];
+    const newConfig: IChartConfig = { ...config };
+
+    const measuresItems = findBucketByLocalIdentifier(newBuckets, MEASURES).items;
+    const measuresCount = measuresItems.length;
+
+    if (measuresCount === 1) {
+        const isComputeRatio = getComputeRatio(measuresItems[0]);
+        const hasStackByAttribute = isStackedChart(buckets);
+        if (isComputeRatio && newConfig.stackMeasures) {
+            newConfig.stackMeasures = false;
+        }
+        if (isComputeRatio || !hasStackByAttribute) {
+            if (newConfig.stackMeasuresToPercent) {
```

Comment:
I think it would be nice to combine 2 if statements into one

Reply:
updated